cognite/client/_api/sequences.py

```diff
@@ -185,6 +186,72 @@ def list(
+                return "LONG"
+            elif np.issubdtype(t, np.number):
+                return "DOUBLE"
+            else:
+                return "STRING"
+
+        return [{"externalId": col, "name": col, "valueType": map_type(t)} for col, t in dataframe.dtypes.iteritems()]
+
+    def create_from_dataframe(
+        self,
+        dataframe: "pandas.DataFrame",
+        external_id: str,
+        force_recreate: bool = False,
+        clear_existing_data: bool = True,
+        *args
+    ):
+        """`Create and populate a sequence from a pandas dataframe . <https://docs.cognite.com/api/v1/#operation/createSequence>`_
+
+                Args:
+                    dataframe (pandas.DataFrame): the data to insert. column names will be used for column name and externalId, column dtypes will be used to determine valueType.
+                    external_id (str): external id of the sequence.
+                    force_recreate (bool): if sequence exists and column types do not match, re-create the sequence.
+                    clear_existing_data (bool): Delete rows that existed before and are not in the given dataframe (i.e. ensures the sequences equals the dataframe). If False, will merge with existing data.
+                    *args: additional arguments to be passed to new Sequence object (description, metadata, etc.)
+
+                Returns:
+                    Sequence: The created or existing sequence.
+
+                Examples:
+
+                    Create a new sequence from a csv file::
+
+                        >>> from cognite.client import CogniteClient
+                        >>> c = CogniteClient()
+                        >>> df = "your dataframe" # e.g. from pd.read_csv("my_csv_file.csv")
+                        >>> seq = c.sequences.create_from_dataframe(df,force_recreate=True)
+                """
+        cols = self._dataframe_to_columns_definition(dataframe=dataframe)
+        existing_seq = self.retrieve(external_id=external_id)
+        if existing_seq:
+            existing_cols_map = {col["externalId"]: col["valueType"] for col in existing_seq.columns}
+            df_cols_map = {col["externalId"]: col["valueType"] for col in cols}
+            if existing_cols_map != df_cols_map:
+                if not force_recreate:
+                    raise IncompatibleColumnTypesError(
+                        "Sequence exists, column types {} do not match dataframe columns {}, and force_recreate=False".format(
+                            existing_cols_map, df_cols_map
+                        )
+                    )
+                else:
+                    self.delete(external_id=external_id)
+                    existing_seq = None
+            else:
+                if clear_existing_data:
+                    self.data.delete_range(external_id=external_id, start=0, end=None)
+
+        if not existing_seq:
+            created_seq = self.create(Sequence(external_id=external_id, columns=cols, *args))
```

Comment:
You cannot specify args after kwargs in Python, so this will fail.

Reply:
you're right, should be kwargs, fixed